controllers/cloudstackmachine_controller.go

```diff
@@ -179,6 +181,17 @@ func (r *CloudStackMachineReconciliationRunner) GetOrCreateVMInstance() (retRes
 	return ctrl.Result{}, err
 }
 
+func processUserData(data []byte, r *CloudStackMachineReconciliationRunner) string {
+	// since cloudstack metadata does not allow custom data added into meta_data, following line is a hack to specify a hostname name
+	// {{ ds.meta_data.hostname }} is expected to be used as a name when kubelet register a node
+	// if more custom data needed to injected, this can be refactored into a method -- processCustomMetadata()
+	hostnameMatcher := regexp.MustCompile(`\{\{\s*ds\.meta_data\.hostname\s*\}\}`)
+	failuredomainMatcher := regexp.MustCompile(`\{\{\s*ds\.meta_data\.failuredomain\s*\}\}`)
```

Comment:
could we make these constants outside the method? I think if there are regex issues we want to know about it at build time rather than runtime